### PR TITLE
Support hybrid test environment

### PR DESCRIPTION
**Pull request by psathyan**
This PR adds support for using a virtual OCP environment and having OpenStack computes on physical environments. The changes that are done here

- Supports extra networks created by `ci_nmstate`
- Performances all the necessary configurations to enable using of the extra networks
- Enables of reading of physical nodes

As a pull request owner and reviewers, I checked that:
- [x] Appropriate testing is done and actually running

### CODE REVIEW

File: ci_framework/roles/devscripts/tasks/cleanup.yml [hunk 1]
```diff
@@ -54,7 +54,11 @@
 - name: Cleanup the devscripts working directory.
   become: true
   ansible.builtin.file:
-    path: "{{ cifmw_devscripts_config.working_dir }}"
+    path: >-
+      {{
+        cifmw_devscripts_config.working_dir |
+        default('/home/dev-scripts')
```
Comment (cjeanner) [inline ci_framework/roles/devscripts/tasks/cleanup.yml:60]: I thought it was set in the defaults already?

Comment (psathyan) [reply ci_framework/roles/devscripts/tasks/cleanup.yml]: Yes, it is configured as `defaults`... thanks looks like I was cautious

**Pull request merged**